File: QMCB-be/app/dto/truth_table.py
from dataclasses import dataclass, asdict, field


@dataclass
class TruthTableDTO:
    input: list[str]
    output: list[str]
    probabilities: list[list[float]] = field(default_factory=list)
    amplitudes: list[list[list[float]]] = field(default_factory=list)

    def to_dict(self):
        return asdict(self)

File: QMCB-fe/src/interfaces/truthTable.ts
/**
 * DTO: shape of a truth table returned by the backend.
 */

export interface TruthTableDTO {
  input: readonly string[];
  output: readonly string[];
  probabilities?: readonly (readonly number[])[];
  amplitudes?: readonly (readonly [number, number])[][];
}

/** One row passed from the controller to the display layer. */
export interface TruthRow {
  input: string;
  trial: string;
  target: string;
  ok: boolean;
  trialProbabilities?: readonly number[];
  targetProbabilities?: readonly number[];
  amplitudes?: readonly (readonly [number, number])[];
}
